test/jira-operations-test.js

```diff
@@ -35,6 +39,31 @@ describe('JIRA Operations Tests', function() {
         });
     });
 
+    it('Find parent from story with Sub-task parent', () => {
+      return findParent(dummyJira.issues.LinkedStory1, dummyJira)
+        .then(parent => {
+          parent.fields.issuetype.name.should.eql('Sub-task');
+        });
+    });
+
+    it('Find parent from story with Initiative, Epic, and Sub-task parents', () => {
+      return findParent(dummyJira.issues.LinkedStory2, dummyJira).then( parent => {
+        parent.fields.issuetype.name.should.eql('Initiative');
+      });
+    });
+
+    it('Find parent from story with Epic and Sub-task parents', () => {
+      return findParent(dummyJira.issues.LinkedStory3, dummyJira)
+        .then(parent => {
+          parent.fields.issuetype.name.should.eql('Epic');
+        });
+    });
+
+    it('Find parent\'s parent of a Sub-task that is under a story with a link to a Dispatcher Sub-task', ()=> {
+      findParent(dummyJira.issues.LinkedSubtask1, dummyJira)
+        .then(subtaskParent => findParent(subtaskParent, dummyJira).then(storyParent => findParent(storyParent, dummyJira).should.eventually.have.deep.property('fields.issuetype.name', 'Dispatcher')));
```

Comment:
Another note on why this is useful. Since we were in grade school we've been taught to read from top to bottom first then left to right. It's always easier to read code vertically then horizontally. It's less of a mental burden in the long run to consume code with this optimization.